sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java

```diff
@@ -538,9 +556,8 @@ public void renewMessageLock(ServiceBusReceivedMessage message, Duration maxLock
      * Sets the state of the session if this receiver is a session receiver.
      *
      * @return The next expiration time for the session lock.
-     * @throws NullPointerException if {@code sessionId} is null.
-     * @throws IllegalArgumentException if {@code sessionId} is an empty string.
      * @throws IllegalStateException if the receiver is a non-session receiver or receiver is already disposed.
+     * @throws ServiceBusException if the session lock cannot be renewed.
      */
```

Comment:
`if the session lock cannot be renewed.` : We have this line many places, but it does not give hint why this operation can not be done. One reason is  id client is already disposed. Should we let user know `why`  ?

Reply:
If the client is disposed, it'll throw an IllegalStateException. Any number of AMQP Exceptions can occur. These are all wrapped into a ServiceBusException.